README.md

```diff
@@ -1,2 +1,167 @@
+A DP project is run for each batch and the locations and images for each DP run are saved as intermediate files.
+The larger the batch size, the faster a stream will complete (DP will not need to be initialized as often).
+However, a larger batch size also corresponds to more intermediate files and more memory being allocated at once by DP (step 3) and PyCytominer (step 4).
+
+Once a stream is set up, it processes each batch as follows:
+1) Download a movie to temporary directory (step 0).
+2) Find desired frame(s) and perform illumination correction.
+Save desired frame(s) in DP project (step 1).
+3) Find nuclei locations in desired frame(s) and save these nuclei locations in DP project (step 2).
+4) Compile an `index.csv` file for the batch.
+Each DP run needs this file to understand the metadata (plate, well, pertubation) associated with the image.
+Save `index.csv` file in DP project (step 2.5).
+5) Profile all images in batch with DeepProfiler.
+Features from this DP run are saved in DP project (step 3).
+6) Compile features from a batch and associate the metadata with features.
+Compiled features with metadata are saved to the final data directory (step 4).
+7) Delete all intermediate files from batch run and run next batch!
+
+## Setup
+
+### Necessary Packages:
+
+The necessary packages for idrstream can be installed into a conda environment with the following:
+```sh
+# Run this command to create the conda environment for idrstream
+conda env create -f idrstream_env.yml
+```
+
+This environment must be activated before using `idrstream` with the following:
+```sh
+# Run this command to activate the conda environment for idrstream
+conda activate idrstream
+```
+
+### Aspera Downloader:
+
+#### Step 1: Install Aspera
+
+Install Aspera high-speed transfer client as described at https://github.com/IBM/aspera-cli#installation.
+We used the direct installation method.
+
+##### Step 1a: Allow Aspera to run without password
+
+It is necessary to allow Aspera command line interface (ascp) to run without a password to use it more easily with python.
+In order to run ascp without a password do the following:
+1) Type sudo visudo at the terminal to open the sudo permissions (sudoers) file
+2) Find the line with: `%sudo   ALL=(ALL:ALL) ALL`
+3) **Below that line**, insert the following line: `username  ALL=(ALL) NOPASSWD: path_to_aspera`
+
+  Note: `username` is your username and path_to_aspera can be found with `ascli config ascp show`
+  
+4) Save and exit the editor
+
+#### Step 2: Download Aspera Public Key
+
+Download and configure [Aspera public key](https://idr.openmicroscopy.org/about/img/aspera/asperaweb_id_dsa.openssh).
+
+### ImageJ:
+
+PyImageJ needs to use be initialized with [FIJI](https://imagej.net/software/fiji/) to be able to read downloaded mitosis movies (CH5 files).
+Download FIJI from the [downloads page](https://imagej.net/software/fiji/downloads) and setup Fiji.app.
+
+### PyBaSiC Illumination Correction Preprocessor:
```

Comment:
Can you add a brief comment about the illumination correction strategy? I think the comment would add clarity about how your approach departs from traditional illumination correction

Reply:
Done in https://github.com/WayScience/IDR_stream/pull/2/commits/3ea498261fec56d57cb0f418942f6176916345ab